packages/ui/src/app/GlobalModals.tsx

```diff
@@ -205,6 +205,12 @@ export const GlobalModals = () => {
   const { status } = useTransactionStatus()
   const Modal = useMemo(() => (modal && modal in modals ? memo(() => modals[modal as ModalNames]) : null), [modal])
 
+  const [container, setContainer] = useState(document.body)
+  useEffect(() => {
+    const container = document.getElementById('modal-container')
+    if (container) setContainer(container)
```

Comment:
It's clear that the `useEffect` context defines its own `container` to set and overwrite the outer one to the selected element in the document tree.